RakutenSearch/Presentation/ViewModel/NetShoppingViewModel.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  NetShoppingViewModel.swift
+//  RakutenSearch
+//
+//  Created by 近藤大伍 on 2023/02/21.
+//
+
+import Foundation
+import Combine
+
+
+protocol NetShoppingViewModelable {
+    var showWebViewPublisher: AnyPublisher<URL, Never> { get }
+    var onSearchButtonClickedPublisher: AnyPublisher<Void, Never> { get }
+    var products:[Product] { get }
+    func fetch(query: String?) async
+    func handleDidSelectRowAt(_ indexPath: IndexPath)
+}
+
+final class NetShoppingViewModel {
+    private var showWebViewSubject = PassthroughSubject<URL, Never>()
+    private var onSearchButtonClickedSubject = PassthroughSubject<Void, Never>()
+    var products:[Product] = []
+    
+    var showWebViewPublisher: AnyPublisher<URL, Never> {
+        return showWebViewSubject.eraseToAnyPublisher()
+    }
+    
+    var onSearchButtonClickedPublisher: AnyPublisher<Void, Never> {
+        return onSearchButtonClickedSubject.eraseToAnyPublisher()
+    }
+    
+    private let productRepository: ProductRepository
+    
+    convenience init() {
+        self.init(repository: RakutenProductRepository())
+    }
+    
+    private init(repository: ProductRepository) {
+        productRepository = repository
+    }
```

Comment:
```suggestion
    private let productRepository = RakutenProductRepository()
```

特に意味がないなら、↑これでinit()要らないと思ったけど、なんか参考にしたサイトとかある？

Reply:
これも上のやつと同じサイト参考にしたんだよね
てか基本これ参考にしてる

記事　https://www.fuwamaki.com/article/255
git　　https://github.dev/fuwamaki/sample-combine-uikit/blob/main/Sample/ViewModel/MainViewModel.swift